lib/src/stack/coap_reliability_layer.dart

```diff
@@ -128,24 +123,7 @@ class CoapReliabilityLayer extends CoapAbstractLayer {
     final CoapResponse? response,
   ) {
     final mt = response!.type;
-    if (mt == CoapMessageType.unknown) {
```

Comment:
Haven't read all of the code and a little sleepy. When would this happen, if the server sent an invalid response and/or if the response was corrupted in flight? I take it with your change we won't acknowledge that message then, meaning that the server will resend the response? Sounds reasonable